src/protobuf-net.Grpc.AspNetCore/ServicesExtensions.cs

```diff
@@ -30,6 +31,21 @@ public static IGrpcServerBuilder AddCodeFirstGrpc(this IServiceCollection servic
             return builder;
         }
 
+        public static IGrpcServerBuilder WithBinderConfiguration(this IGrpcServerBuilder builder, Action<BinderOptions> binderOptions)
+        {
+            var options = new BinderOptions();
```

Comment:
note that *as I understand it* (I'm not a DI expect), this not the normal pattern for "option" scenarios; not sure whether that matters much - this will probably *work*, although it may have atypical timing in the DI pipe

Reply:
Makes sense. I'll switch to your `AddCodeFirstGrpc` suggestion as it fixes both issues.

Reply:
It's not an options pattern. but it is fairly common when the config object passed in is not just an options object to be registered, but is actually used inside to configure how things are registered with DI.
I was going to name it `BinderConfiguration`, but that is what the main one is called.

Reply:
Note it doesn't affect the DI pipeline either as everything is still registered/created explicitly as I invoke the `BinderOptions()` so it is all in place before the serviceprovider is built and requests made of it.

Reply:
Or if you were meaning that I should have been wrapping in a "value factory"
```cs
builder.Services.TryAddSingleton(() => BinderConfiguration.Create(options.MarshallerFactories, options.ServiceBinder));
```
then I agree.